artemis-server/src/main/java/org/apache/activemq/artemis/core/management/impl/view/predicate/ConsumerFilterPredicate.java

```diff
@@ -59,6 +62,28 @@ public boolean test(ServerConsumer consumer) {
             return matches(server.getSessionByID(consumer.getSessionID()).getRemotingConnection().getTransportConnection().getLocalAddress());
          case REMOTE_ADDRESS:
             return matches(server.getSessionByID(consumer.getSessionID()).getRemotingConnection().getTransportConnection().getRemoteAddress());
+         case DELIVERING_COUNT:
+            return matches(consumer.getDeliveringMessages().size());
+         case DELIVERING_SIZE:
+            List<MessageReference> deliveringMessages = consumer.getDeliveringMessages();
+            long deliveringMessageSize = 0;
```

Comment:
Ditto other comment i would be tracking this at the consumerimpl as messages added or acl'd the size is added and removed so were not later looping, and like wise avoid concurrent issues